libmuscle/cpp/src/libmuscle/mmp_client.cpp

```diff
@@ -64,12 +66,41 @@ namespace {
         return Data::list(std::string(port.name), encode_operator(port.oper));
     }
 
+    template <class T>
+    Data encode_optional(Optional<T> const & value) {
+        Data encoded;
+        if (value.is_set())
+            encoded = value.get();
+        return encoded;
+    }
```

Comment:
Sounds like a good idea, but looking at it again, we're really only encoding stuff when sending it to the manager or to a peer, and the peer data format doesn't support optional types (or everything is optional, since we have a nil value, anyway...). So I'm not sure where else this would be needed, I think I'd rather keep it with the other encoding functions here. We can always pull it out later if we need it elsewhere.